src/providers/SettingContext.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { PropsWithChildren, ReactElement, createContext } from "react"
+import { darkTheme, lightTheme } from "../theme"
+import { CssBaseline } from "@mui/material"
+import { ThemeProvider } from "@mui/material"
+import { getCookie } from "../utils/helper"
+import useMediaQuery from "@mui/material/useMediaQuery"
+
+export type ThemeMode = "light" | "dark" | "system"
+export type SettingsContextProps = {
+  themeMode: ThemeMode | null
+  onChangeMode: (themeMode: ThemeMode) => void
+}
+
+const initialState: SettingsContextProps = {
+  themeMode: "light",
+  onChangeMode: () => undefined,
+}
+
+const SettingsContext = createContext(initialState)
+
+function SettingsProvider({
+  children,
+}: PropsWithChildren<unknown>): ReactElement {
+  const [mode, setMode] = React.useState<ThemeMode | null>(null)
+  const prefersDarkMode = (useMediaQuery("(prefers-color-scheme: dark)")
```

Comment:
This should be a boolean `isSystemDarkMode = useMediaQuery("(prefers-color-scheme: dark)")`. 
The user can still manually choose light mode even if their system is set to dark mode, so we need to choose the correct one. 

See how uniswap does this here:
Keep their internal state up to date with the system: https://github.com/Uniswap/interface/blob/main/src/state/user/updater.tsx#L10
Use their internal state to choose correct theme:  https://github.com/Uniswap/interface/blob/main/src/state/user/hooks.tsx#L65

Reply:
I think now it is working as you want.
Could you please check the following video?

https://user-images.githubusercontent.com/75422800/146211947-a310ec3b-94a2-4f45-b23b-82951f868668.mov

The reason I embed `system` variant on `ThemeMode` is for future. If we could have 3 kinds(`system` `light` `dark`) of  setting for theme mode instead of `light` and `dark` on the UI, then we can use this code without change.